homeassistant/components/zwave_js/addon.py

```diff
@@ -0,0 +1,256 @@
+            addon_info: dict = await async_get_addon_info(self._hass, ADDON_SLUG)
+        except HassioAPIError as err:
+            raise AddonError("Failed to get the Z-Wave JS add-on info") from err
+
+        return addon_info
+
+    async def async_is_addon_running(self) -> bool:
+        """Return True if Z-Wave JS add-on is running."""
+        addon_info = await self.async_get_addon_info()
+        return bool(addon_info["state"] == "started")
+
+    async def async_is_addon_installed(self) -> bool:
+        """Return True if Z-Wave JS add-on is installed."""
+        addon_info = await self.async_get_addon_info()
+        return addon_info["version"] is not None
+
+    async def async_get_addon_options(self) -> dict:
+        """Get Z-Wave JS add-on options."""
+        addon_info = await self.async_get_addon_info()
+        return cast(dict, addon_info["options"])
+
+    async def async_set_addon_options(self, config: dict) -> None:
+        """Set Z-Wave JS add-on options."""
+        options = {"options": config}
+        try:
+            await async_set_addon_options(self._hass, ADDON_SLUG, options)
+        except HassioAPIError as err:
+            raise AddonError("Failed to set the Z-Wave JS add-on options") from err
+
+    async def async_install_addon(self) -> None:
+        """Install the Z-Wave JS add-on."""
+        try:
+            await async_install_addon(self._hass, ADDON_SLUG)
+        except HassioAPIError as err:
+            raise AddonError("Failed to install the Z-Wave JS add-on") from err
+
+    @callback
+    def async_schedule_install_addon(
+        self, usb_path: str, network_key: str
+    ) -> asyncio.Task:
+        """Schedule a task that installs and sets up the Z-Wave JS add-on.
+
+        Only schedule a new install task if the there's no running task.
+        """
+        if not self._install_task or self._install_task.done():
+            LOGGER.warning("Z-Wave JS add-on is not installed. Installing add-on")
+            self._install_task = self._async_schedule_addon_operation(
+                self.async_install_addon,
+                partial(self.async_setup_addon, usb_path, network_key),
+            )
+        return self._install_task
+
+    async def async_uninstall_addon(self) -> None:
+        """Uninstall the Z-Wave JS add-on."""
+        try:
+            await async_uninstall_addon(self._hass, ADDON_SLUG)
+        except HassioAPIError as err:
+            raise AddonError("Failed to uninstall the Z-Wave JS add-on") from err
+
+    async def async_update_addon(self) -> None:
+        """Update the Z-Wave JS add-on if needed."""
+        addon_info = await self.async_get_addon_info()
+        addon_version = addon_info["version"]
+        addon_latest_version = addon_info["version_latest"]
+
+        if addon_version is None:
+            raise AddonError("Z-Wave JS add-on is not installed")
+
+        if AwesomeVersion(addon_version) >= AwesomeVersion(addon_latest_version):
```

Comment:
I don't think that you need to compare but just check if they are different to see if an update is available. CC @ludeeus

Reply:
Addons have a `update_available` boolean in the dictionary you can/should use instead